table/scoped_arena_iterator.h

```diff
@@ -8,31 +8,43 @@
 #pragma once
 
 #include "port/port.h"
+#include "rocksdb/iterator.h"
 #include "table/internal_iterator.h"
 
 namespace ROCKSDB_NAMESPACE {
-class ScopedArenaIterator {
-  void reset(InternalIterator* iter) noexcept {
+// A wrapper iterator used to wrap an `InternalIterator` allocated on an arena.
+// It also supports wrapping an `InternalIterator` allocated with the default
+// allocator. For this use case, is_arena_mode should be set to false explicitly
+// when constructing the `ScopedArenaIterator`.
+class ScopedArenaIterator : public Iterator {
```

Comment:
I don't like reusing ScopedArenaIterator in this way. My objection is mostly about the existing state of ScopedArenaIterator, so I'd like you to rebase on #12470 (once approved ;) ) and do this wrapping some other way, which I'm confident will be fine (even if you create a new class with most of the same code you've added here).

Reply:
Thank you for refactoring `ScopedArenaIterator` to what it should have been. I have added a `TableIterator` class as a wrapper iterator class to use for this API.